ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -673,11 +583,26 @@ private ExecuteResult execute(final PreparedStatement<?> statement) {
       return PreparedStatement.of(statement.getStatementText(), query);
     }
 
-    private static PreparedStatement<?> postProcessInsertIntoStatement(
+    private PreparedStatement<?> preProcessInsertIntoStatement(
         final PreparedStatement<?> statement
     ) {
       final InsertInto insertInto = (InsertInto) statement.getStatement();
 
+      final String targetName = insertInto.getTarget().getSuffix();
```

Comment:
Check to ensure we're not `INSERT INTO` a table moved from the `parseStatements` call, (where it didn't work), and into the `execute` call, were it does work.